adr-11-centralized-sequencer.md

```diff
@@ -0,0 +1,221 @@
+       TransactionStatus   metrics.Counter  // Counts transaction status outcomes
+       NumPendingBlocks    metrics.Gauge    // Tracks the number of blocks waiting to be submitted
+       IncludedBlockHeight metrics.Gauge    // Tracks the height of the last included block in the DA layer
+   }
+   ```
+
+### APIs
+
+The centralized sequencer implements the Generic Sequencer interface from the `go-sequencing` package:
+
+```go
+type Sequencer interface {
+    SubmitRollupBatchTxs(ctx context.Context, req SubmitRollupBatchTxsRequest) (*SubmitRollupBatchTxsResponse, error)
+    GetNextBatch(ctx context.Context, req GetNextBatchRequest) (*GetNextBatchResponse, error)
+    VerifyBatch(ctx context.Context, req VerifyBatchRequest) (*VerifyBatchResponse, error)
+}
+```
+
+These methods are exposed via a gRPC service that rollup clients can interact with.
+The centralized sequencer implements the following methods as part of the Generic Sequencer interface:
+
+1. **SubmitRollupBatchTxs**:
+   - This method is responsible for accepting a batch of transactions from a rollup client. It takes a context and a request containing the rollup ID and the batch of transactions to be submitted.
+   - The method first validates the rollup ID to ensure it matches the expected ID for the sequencer. If the ID is invalid, it returns an error.
+   - Upon successful validation, the method adds the transactions to the internal transaction queue (`TransactionQueue`) for processing.
+   - It then triggers the batch submission process, which involves retrieving the next batch of transactions and submitting them to the designated Data Availability (DA) layer.
+   - Finally, it returns a response indicating the success or failure of the submission.
+
+2. **GetNextBatch**:
+   - This method retrieves the next batch of transactions that are ready to be processed by the rollup. It takes a context and a request containing the rollup ID and the last batch hash.
+   - The method first checks if the rollup ID is valid. If not, it returns an error.
+   - It then verifies the last batch hash to ensure that the rollup client is requesting the correct next batch.
+   - If a valid batch is found, it prepares the batch response, which includes the batch of transactions and a timestamp.
+   - If no transactions are available, it returns an empty batch response.
+
+3. **VerifyBatch**:
+   - This method is used to verify the that a batch received from the sequencer was actually published on the DA layer. It takes a context and a request containing the rollup ID and the batch hash.
+   - Similar to the other methods, it first validates the rollup ID.
+   - It then checks if the provided batch hash exists in the internal data structure that tracks seen batches.
+   - If the batch hash is found, it returns a response indicating that the batch is valid. If not, it returns a response indicating that the batch is invalid.
+
```

Comment:
should we add a TODO here? Currently VerifyBatch only does a quick verification of the batch hash exist in the seenBatches, however in the future, it is expected to perform the full DA verification where the batch submitted to DA as referenced using a DA height and VerifyBatch would query the DA at the referenced height to verify that the batch exists in DA.

Reply:
would this be what the pending header system would work off of.

Reply:
pending headers in rollkit is different. the verifybatch is for transaction batches. 
verifyBatch for the batch producer (which is rollkit's header producer) is basically `return true`
for fullnodes, it needs to provide a stateless verification, meaning the verification is done using the DA